plugin.py

```diff
@@ -79,27 +144,191 @@ class TellRecord(Base):
 # session.rollback()
 session = DBSession()
 
+
 class Tell(callbacks.Plugin):
     """MemoServ replacement with extra features."""
     threaded = True
 
+    queryTell = PostTell()
+
+    pm = 0
+
+    def __init__(self, irc):
+        self.__parent = super(Tell, self)
+        self.__parent.__init__(irc)
+
+        # TODO: Load all unread messages.
+        self.queryTell.load_unread()
+
+    # Process all text before handing off to command processor
+    def inFilter(self, irc, msg):
+        # Return days, hours, minutes ago with *simple* logic
+        def get_timeago():
+            return humanize.naturaltime(datetime.datetime.now() - t['time'])
+
+        # If !delaytells is the command, obviously don't relay tells.
+        if str(msg).find('!delaytells') != -1 or str(msg).find('!skiptells') != -1:
+            return msg
+
+        # Attempt to query any past tells for the user.
+        if msg.command == "PRIVMSG":
+            tells = self.queryTell.query_post(msg.nick)
+            _channel = msg.args[0]
+
+            # Sending PM to bot?
+            if _channel == irc.nick:
+                self.pm = 1
+            else:
+                # Check that we have a valid channel
+                if not ircutils.isChannel(_channel):
+                    return msg
+                self.pm = 0
+
+            # Process any tells for the user when they type
+            if tells is not None:
+                # First check if we have a delay.
+                if tells['delay'] is not None:
+                    _now = datetime.datetime.now()
+                    try:
+                        if _now < tells['delay']:
+                            # Don't relay tells. We haven't passed the delay time
+                            return msg
+                    except:
+                        pass
+
+                _public = self.registryValue('youhavemail')
+                _private = self.registryValue('youhaveprivatemail')
+                _priv_tells = []
+                _pub_tells = []
+                _relay_pub = False
+                _relay_private = False
+                _priv_count = 0
+                _pub_count = 0
+
+                # Format and divvy out private and public tells.
+                for t in tells['tells']:
+                    # Set the Tell to read
+                    _id = t['id']
+                    self.queryTell.message_read(_id, msg.nick)
+                    # This will tie to config, tellMessage. If you want to add more variables, do it here.
+                    _d = {'time_ago': get_timeago(), 'from': t['from'], 'content': t['content']}
+                    if t['private'] is True:
+                        _priv_count += 1
+                        _relay_private = True
+                        _priv_tells.append(self.registryValue("tellMessage").format(**_d))
```

Comment:
Store `self.registryValue("tellMessage")` above in a variable, DRY. Now that you have to change the config name you can see that you have to change it twice. If you have the suspicion that you're duplicating code, even if you think it may never change you should still move it into a member.